samtranslator/swagger/swagger.py

```diff
@@ -92,12 +116,15 @@ def add_lambda_integration(self, path, method, integration_uri):
         """
 
         method = self._normalize_method_name(method)
-
         if self.has_integration(path, method):
             raise ValueError("Lambda integration already exists on Path={}, Method={}".format(path, method))
 
         self.add_path(path, method)
 
+        # Wrap the integration_uri in a Condition if one exists on that function:
+        if condition:
+            integration_uri = make_conditional(condition, integration_uri)
```

Comment:
This is necessary, or else CFN complains that it can't resolve a resource reference.

Reply:
Sample output:

The contents of the method (in this case, **post** are wrapped in a `Fn::If` conditional. CFN doesn't recognize that the `uri` field is wrapped in this conditional (creates an "unresolved resource reference" error), so another conditional is needed around that particular section. This is why we need two conditionals for one template block.

```json
"paths": {
            "/": {
              "post": {
                "Fn::If": [
                  "MyCondition", 
                  {
                    "x-amazon-apigateway-integration": {
                      "httpMethod": "POST", 
                      "type": "aws_proxy", 
                      "uri": {
                        "Fn::If": [
                          "MyCondition", 
                          {
                            "Fn::Sub": "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${ImplicitApiFunction.Arn}/invocations"
                          }, 
                          {
                            "Ref": "AWS::NoValue"
                          }
                        ]
                      }
                    }, 
                    "responses": {}
                  }, 
                  {
                    "Ref": "AWS::NoValue"
                  }
                ]
              }, 
```